job-dsls/jobs/tools_provisioning.groovy

```diff
@@ -0,0 +1,428 @@
+                        [name: "DDNS_HASH", default: "efbc8e76038362a51614055313907d5a", description: "Hash used as authorization for use of specified hostname." ]
+                ]
+        ],
+        "provision-smee-client"        : [
+                jobAbr: "SmeeClient",
+                params : [
+                        [name: "IMAGE", default: "rhel-7.6-server-x86_64-released", description: "The name of the image to be used for machine creation." ],
+                        [name: "FLAVOUR", default: "ci.m1.medium.no.nested.virt", description: "The flavor (i.e. resources such as CPU cores, RAM, ...) defining the machine. m1.medium = 2 vCPUs, 4 GB RAM, 40 GB HDD" ],
+                        [name: "WEBHOOK_URL", default: "https://smee.io/fnCWbn1nllLyJG", description: ""],
+                        [name: "TARGET_URL", default: "https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/ghprbhook/", description: ""]
+                ]
+         ],
+        "provision-verdaccio-service"  : [
+                jobAbr: "VerdaccioServ",
+                params : [
+                        [name: "IMAGE", default: "rhel-7.6-server-x86_64-released", description: "The name of the image to be used for machine creation." ],
+                        [name: "FLAVOUR", default: "ci.m1.medium.no.nested.virt", description: "The flavor (i.e. resources such as CPU cores, RAM, ...) defining the machine. m1.medium = 2 vCPUs, 4 GB RAM, 40 GB HDD" ]
+                ]
+        ]
+]
+
+//create folders
+folder("Provisioning")
+
+for (jobNames in JOB_NAMES) {
+    Closure<Object> get = { String key -> jobNames.value[key] ?: DEFAULTS[key] }
+
+    String jobName = jobNames.key
+    String folderPath = get("folderPath")
+    String jobAbr = get("jobAbr")
+    String labExp = get("labExp")
+    String openJdk = get("openJdk")
+    def params = get("params")
+    def logRot = get("logRot")
+    def timeOutVar = get("timeOutVar")
+
+    // jobs for master branch don't use the branch in the name
+    String jobN = "$folderPath/$jobName"
+
+    job(jobN) {
+        description(getDescription(jobAbr))
+
+        logRotator {
+            numToKeep(logRot)
+        }
+
+        label(labExp)
+
+        jdk(openJdk)
+
+        for ( param in params ) {
+            parameters {
+                stringParam(param.name, param.default, param.description)
+                if ( jobAbr == 'VerdaccioServ') {
+                    password{
+                        name('ENCRYPTED_PASSWORD')
+                        defaultValue('redhat')
+                        description('The encrypted password. To generate it you should take value from npm adduser command. Default redhat')
+                    }
+                }
+            }
+        }
+
+        // Allows a job to check out sources from an SCM provider.
+        scm {
+            git {
+                if ( jobAbr == 'CekitCacher' || jobAbr == 'DockerReg' ) {
```

Comment:
I would consider to split this DSL and create a single one for every job to avoid too many specific conditions based on single jobs. Some repetition is not that bad.